torch/_dynamo/variables/builder.py

```diff
@@ -692,7 +693,9 @@ def build_key_value(i, k, v):
             items = [SourcelessBuilder.create(self.tx, v) for v in value]
             self.install_guards(GuardBuilder.ID_MATCH)
             return FrozensetVariable(items, source=self.source)
-        elif isinstance(value, (enum.Enum, torch.DispatchKey)):
+        elif isinstance(
+            value, (enum.Enum, torch.DispatchKey, torch._C._functorch.TransformType)
+        ):
```

Comment:
Are we sure that DispatchKey and TransformType actually behave like Python enums? (I don't care if they don't, this is close enough, but I'm just curious)

Reply:
If they do then I want some generic way for Dynamo to support these C++ enums. Maybe shove them into a list somewhere near EnumVariable

Reply:
I think so, as `DispatchKey` and `TransformType` are pure python classes extending from `enum.Enum`, even though they are bound to their corresponding C++ types.

Reply:
no action required here, might be nice to have an easy(ier) way to add more of these in the future